### PR TITLE
Move fluence-identity to fluence-keypair

### LINKED ISSUE #8: Identity: add method for PeerId generation
Currently users of the library need to explicitly use libp2p:
```rust
use libp2p::identity::ed25519::Keypair;

let kp = Keypair::generate();
let public_key = libp2p::identity::PublicKey::Ed25519(kp.public());
PeerId::from(public_key)
```

fluence-identity's PublicKey (maybe only ed25519, maybe all kinds of keys) must provide api to generate `libp2p::PeerId`

### PR DESCRIPTION
**Pull request by justprosh**
Closes #8 #10

### CODE REVIEW

File: keypair/src/public_key.rs [hunk 1]
```diff
@@ -99,21 +107,58 @@ impl PublicKey {
...
+                libp2p_identity::rsa::PublicKey::decode_x509(&key.encode_x509()).unwrap(),
+            ),
+            PublicKey::Secp256k1(key) => libp2p_identity::PublicKey::Secp256k1(
+                libp2p_identity::secp256k1::PublicKey::decode(&key.encode()[..]).unwrap(),
+            ),
+        }
+    }
+}
+
+pub fn peer_id_to_fluence_pk(peer_id: libp2p_core::PeerId) -> eyre::Result<PublicKey> {
```
Comment (folex) [inline keypair/src/public_key.rs:152]: Why isn't it a `TryFrom`?

File: keypair/src/signature.rs [hunk 2]
```diff
@@ -30,6 +30,11 @@ pub enum Signature {
     Secp256k1(secp256k1::Signature),
 }
 
+pub struct RawSignature {
+    pub bytes: Vec<u8>,
+    pub sig_type: String,
```
Comment (folex) [inline keypair/src/signature.rs:35]: Make `sig_type` a enum plz

File: keypair/src/signature.rs [hunk 3]
```diff
@@ -78,6 +82,41 @@ impl Signature {
...
+        match raw_signature.sig_type.as_str() {
+            "Ed25519" => Ok(Signature::Ed25519(crate::ed25519::Signature(raw_signature.bytes))),
+            #[cfg(not(target_arch = "wasm32"))]
+            "RSA" => Ok(Signature::Rsa(crate::rsa::Signature(raw_signature.bytes))),
+            "Secp256k1" => Ok(Signature::Secp256k1(crate::secp256k1::Signature(raw_signature.bytes))),
+            _ => Err(DecodingError::RawSignatureUnsupportedType(raw_signature.sig_type)),
+        }
+    }
+
+    pub fn from_bytes_with_public_key(pk: &PublicKey, bytes: Vec<u8>) -> Self {
```
Comment (folex) [inline keypair/src/signature.rs:111]: accept here `KeyType` enum instead of a public key

File: keypair/src/signature.rs [hunk 4]
```diff
@@ -78,6 +82,41 @@ impl Signature {
...
+            Rsa(_) => "RSA".to_string(),
+            Secp256k1(_) => "Secp256k1".to_string(),
+        }
+    }
+
+    pub fn get_raw_signature(&self) -> RawSignature {
+        RawSignature { bytes: self.to_vec().clone().to_vec(), sig_type: self.get_signature_type() }
+    }
+
+    pub fn from_raw_signature(raw_signature: RawSignature) -> Result<Self, DecodingError> {
```
Comment (folex) [inline keypair/src/signature.rs:101]: Once you have `KeyType` enum, I think you can remove this method

**Review by folex — APPROVED** [review]

File: keypair/src/public_key.rs [hunk 5]
```diff
@@ -99,21 +107,58 @@ impl PublicKey {
             Secp256k1(pk) => pk.encode().to_vec(),
         }
     }
+
+    pub fn to_peer_id(&self) -> PeerId {
+        PeerId::from_public_key(self.clone().into())
```
[the remark this answers is not in the document]
Comment (justprosh) [reply keypair/src/public_key.rs:112]: I don't know how it can be achieved because peer id inlining protobuf encoded libp2p::identity key

**Review by folex — APPROVED** [review]

**Pull request merged**